### PR TITLE
feat: Create Infrastructure for evaluation and inference around various tasks

### PR DESCRIPTION
**Pull request by soumik12345**
Fixes https://github.com/soumik12345/restorers/issues/30

### CODE REVIEW

File: restorers/evaluation/base.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        unique_weights = {id(w): w for w in weights}.values()
+        # Ignore TrackableWeightHandlers, which will not have a shape defined.
+        unique_weights = [w for w in unique_weights if hasattr(w, "shape")]
+        weight_shapes = [w.shape.as_list() for w in unique_weights]
+        standardized_weight_shapes = [
+            [0 if w_i is None else w_i for w_i in w] for w in weight_shapes
+        ]
+        return int(sum(np.prod(p) for p in standardized_weight_shapes))
+
+    def get_gflops(self) -> float:
```
Comment (soumik12345) [inline restorers/evaluation/base.py:66]: @ariG23498 @SauravMaheshkar The evaluation infrastructure is ready. However, I was unable to get this particular function working. Can you guys please lend some help?

Workspace for reference: https://wandb.ai/ml-colabs/zero-dce

File: restorers/dataloader/base/commons.py [hunk 2]
```diff
@@ -21,3 +21,10 @@ def random_vertical_flip(low_image, enhanced_image):
             tf.image.flip_up_down(enhanced_image),
         ),
     )
+
+
+def read_image(image_path: str, normalization_factor) -> tf.Tensor:
+    image = tf.io.read_file(image_path)
+    image = tf.io.decode_image(image, channels=3, expand_animations=False)
+    image = tf.cast(image, dtype=tf.float32) / normalization_factor
```
Comment (ariG23498) [inline restorers/dataloader/base/commons.py:29]: Should reading an image contain normalization code in it?

This is not a blocker -- I just wanted to point it out.

**Review by ariG23498 — COMMENTED** [review]
I think the API design is well in place.

Need to look into the [blocking functionality](https://github.com/soumik12345/restorers/pull/33/files?diff=split&w=0#r1096716917) a little more.

Other than that I think we are good to go on this one!

File: restorers/dataloader/base/commons.py (hunk 2 above)
Comment (soumik12345) [reply restorers/dataloader/base/commons.py:29]: We are reading images to be fed into the model and hence the `normalization_fector` is kept here. I think its better to set it to `1` by default.

Comment (ariG23498) [reply restorers/dataloader/base/commons.py:29]: That makes more sense to me! Thanks,

**Review by ariG23498 — APPROVED** [review]
LGTM!

**Comment by soumik12345** [comment]
@ariG23498 made a couple of changes since your last review:

1. Fixed the GFLOPs calculation logic which was previously broken.
2. Aligned the Zero-DCE implementation closer to the original paper and matched its GFLOPs count accurately with the paper.

**Pull request merged**